plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -56,34 +56,30 @@
    * Returns the credential of the active user. If there is no active user, returns {@code null}.
    */
   // Should probably be synchronized properly.
-  // TODO(chanseok): consider returning a String JSON (i.e., hide Credential)
-  public Credential getActiveCredential() throws IOException {
-    GoogleTokenResponse authResponse = logIn();
+  public Credential getActiveCredential(IShellProvider shellProvider) throws IOException {
+    GoogleTokenResponse authResponse = logIn(shellProvider);
     return createCredential(authResponse);
   }
 
-  private GoogleTokenResponse logIn() throws IOException {
-    // 1. Open a browser that takes care of the login.
+  private GoogleTokenResponse logIn(IShellProvider shellProvider) throws IOException {
     GoogleAuthorizationCodeRequestUrl requestUrl = new GoogleAuthorizationCodeRequestUrl(
         OAUTH_CLIENT_ID, GoogleOAuthConstants.OOB_REDIRECT_URI, OAUTH_SCOPES);
 
     try {
       IWorkbenchBrowserSupport browserSupport = PlatformUI.getWorkbench().getBrowserSupport();
       browserSupport.getExternalBrowser().openURL(requestUrl.toURL());
-    } catch (PartInitException pie) {
+    } catch (PartInitException pie) {  // requires org.eclipse.equinox.common OSGi bundle.
```

Comment:
That's right. I expect that `PartInitException` is removed when I create a new login screen, so this is not to forget to remove this dependency later. Sounds fair?